src/main/java/net/neoforged/neoforge/capabilities/Capabilities.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.capabilities;
+
+import net.minecraft.core.Direction;
+import net.minecraft.resources.ResourceLocation;
+import net.neoforged.neoforge.energy.IEnergyStorage;
+import net.neoforged.neoforge.fluids.capability.IFluidHandler;
+import net.neoforged.neoforge.fluids.capability.IFluidHandlerItem;
+import net.neoforged.neoforge.items.IItemHandler;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Capabilities provided by NeoForge itself, for modders to directly reference.
+ */
+public class Capabilities {
+    public static class EnergyStorage {
+        public static final BlockCapability<IEnergyStorage, @Nullable Direction> BLOCK = BlockCapability.create(create("energy"), IEnergyStorage.class, Direction.class);
+        public static final EntityCapability<IEnergyStorage, Void> ENTITY = EntityCapability.createVoid(create("energy"), IEnergyStorage.class);
+        public static final ItemCapability<IEnergyStorage, Void> ITEM = ItemCapability.createVoid(create("energy"), IEnergyStorage.class);
+    }
+
+    public static class FluidHandler {
+        public static final BlockCapability<IFluidHandler, @Nullable Direction> BLOCK = BlockCapability.create(create("fluid_handler"), IFluidHandler.class, Direction.class);
+        public static final EntityCapability<IFluidHandler, Void> ENTITY = EntityCapability.createVoid(create("fluid_handler"), IFluidHandler.class);
+        public static final ItemCapability<IFluidHandlerItem, Void> ITEM = ItemCapability.createVoid(create("fluid_handler"), IFluidHandlerItem.class);
+    }
+
+    public static class ItemHandler {
+        public static final BlockCapability<IItemHandler, @Nullable Direction> BLOCK = BlockCapability.create(create("item_handler"), IItemHandler.class, Direction.class);
+        /**
+         * Capability for the inventory of an entity.
+         * If an entity has multiple inventory "subparts", this capability should give a combined view of all the subparts.
+         */
+        public static final EntityCapability<IItemHandler, Void> ENTITY = EntityCapability.createVoid(create("item_handler"), IItemHandler.class);
+        /**
+         * Capability for an inventory of entity that should be accessible to automation,
+         * in the sense that droppers, hoppers, and similar modded devices will try to use it.
+         */
+        public static final EntityCapability<IItemHandler, Void> ENTITY_AUTOMATION = EntityCapability.createVoid(create("item_handler_automation"), IItemHandler.class);
```

Comment:
As discussed on Discord: we should add a side to this capability as well, and pass it when interacting via hopper.

Reply:
Added the nullable direction context for entity automation